addons/_deps/sniffer-poller.js

```diff
@@ -170,23 +192,41 @@ class SnifferPoller {
 			return null;
 		}
 
-		if(!this._prevdata.memoryReadout.arrangementID) {
-			return null;
-		}
-
+		//Look through arrangements for matching ID
 		for (var i = this._prevdata.songDetails.arrangements.length - 1; i >= 0; i--) {
 			var arrangement = this._prevdata.songDetails.arrangements[i];
 
-			if(arrangement.arrangementID == this._prevdata.memoryReadout.arrangementID) {
+			//Check that ID is correctly formatted-32 characters long-to avoid errors
+			if(arrangement.arrangementID.length == 32 && arrangement.arrangementID == this._prevdata.memoryReadout.arrangementID) {
```

Comment:
This check is good, I think it should also be done in [RSMemoryReader](https://github.com/kokolihapihvi/RockSnifferLib/blob/97be10beb2771e63405652daf8d42898bf636f91/RSHelpers/RSMemoryReader.cs#L75)